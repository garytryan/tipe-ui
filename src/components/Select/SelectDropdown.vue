<template>     
  <div 
    class="dropdown" :class="{[size]: true}">
    <div 
      v-for="(option, index) in options"
      :key="option.value"
      :class="{active: index === activeIndex}" 
      tabindex="-1"
      class="dropdown-item"
      @click="$emit('change', option)"><p>{{ option.label }}</p></div>
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import SelectOptionShape from '@/types/SelectOptionShape'

export default {
  name: 'TipeSelectDropdown',
  props: {
    options: vueTypes.arrayOf(vueTypes.shape(SelectOptionShape)),
    size: vueTypes
      .oneOf(['mini', 'small', 'medium', 'large', 'full'])
      .def('full'),
    activeIndex: vueTypes.number
  }
}
</script>

<style scoped lang="postcss">
.dropdown {
  background-color: #fff;
  border-radius: 0 0 5px 5px;
  z-index: 2;
  box-shadow: 5px 20px 40px 0 rgba(0, 0, 0, 0.1);
  padding: 0.6875rem 0;

  &.mini,
  &.small {
    width: 6.5625rem;
  }

  &.medium {
    width: 10rem;
  }

  &.large {
    width: 10.5625rem;
  }

  &.full {
    width: 100%;
  }

  & .dropdown-item {
    height: 2.1875rem;
    display: flex;
    justify-content: center;
    flex-direction: column;

    & p {
      color: #627098;
      font-size: 0.8125rem;
      margin: 0;
      padding-left: 1.0625rem;
    }

    &:hover,
    &.active {
      background-color: #f7f8fb;
      border-left: 3px #1cc5bc solid;

      & p {
        padding-left: calc(1.0625rem - 3px);
      }
    }
  }
}
</style>
