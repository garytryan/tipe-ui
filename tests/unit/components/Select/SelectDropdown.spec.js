import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import SelectDropdown from '@/components/Select/SelectDropdown'

const props = {
  options: [
    { label: 'Owner', value: 'owner' },
    { label: 'Admin', value: 'admin' }
  ],
  activeIndex: 1,
  size: 'large'
}
describe('SelectDropdown.vue', () => {
  it('matches previous snapshot', async () => {
    const wrapper = shallowMount(SelectDropdown, {
      propsData: { ...props }
    })
    await expect(createRenderer().renderToString(wrapper.vm)).toMatchSnapshot()
  })
  it('has correct data-tipe-iu attibute', () => {
    const wrapper = shallowMount(SelectDropdown, {
      propsData: { ...props }
    })
    expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeSelectDropdown')
  })
  it('has dropdown class', () => {
    const wrapper = shallowMount(SelectDropdown, {
      propsData: { ...props }
    })
    expect(wrapper.classes()).toContain('dropdown')
  })
  it('has the correct size class', () => {
    const wrapper = shallowMount(SelectDropdown, {
      propsData: { ...props }
    })
    expect(wrapper.classes()).toContain('large')
  })
  it('should have the correct props', () => {
    const wrapper = shallowMount(SelectDropdown, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.props().options).toBe(props.options)
    expect(wrapper.props().activeIndex).toBe(1)
    expect(wrapper.props().size).toBe('large')
  })
  it('should have the correct number of divs', () => {
    const wrapper = shallowMount(SelectDropdown, {
      propsData: {
        ...props
      }
    })
    const divArr = wrapper.findAll('.dropdown-item')
    expect(divArr).toHaveLength(2)
  })
  it('should have the active class on the second div', () => {
    const wrapper = shallowMount(SelectDropdown, {
      propsData: {
        ...props
      }
    })
    const item = wrapper.findAll('.dropdown-item').at(1)
    expect(item.classes()).toContain('active')
  })
  it('should emit change event', () => {
    const wrapper = shallowMount(SelectDropdown, {
      propsData: {
        ...props
      }
    })
    const itemArr = wrapper.findAll('.dropdown-item')
    itemArr.trigger('click')
    expect(wrapper.emitted().change).toBeTruthy()
  })
})
