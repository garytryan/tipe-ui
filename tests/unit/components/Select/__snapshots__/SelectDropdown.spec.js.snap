// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDropdown.vue matches previous snapshot 1`] = `Promise {}`;
